h/tasks/indexer.py

```diff
@@ -43,3 +47,23 @@ def delete_annotation(id_):
 def sync_annotations(limit):
     search_index = celery.request.find_service(name="search_index")
     search_index.sync(limit)
+
+
+@celery.task(acks_late=False)
+def report_sync_annotations_queue_length():
+    now = datetime.utcnow()
+
+    queue_length = (
+        celery.request.db.query(Job)
+        .filter(
+            Job.name == "sync_annotation",
+            Job.tag.in_(["storage.create_annotation", "storage.update_annotation"]),
+            Job.scheduled_at < now,
+            Job.expires_at > now,
+        )
+        .count()
+    )
+
+    newrelic.agent.record_custom_metric(
+        "Custom/SyncAnnotationsQueueLength", queue_length
```

Comment:
I've changed this to `"Custom/SyncAnnotations/Queue/API/Length"`. Added the `/API/` because this metric is actually only recording those jobs added by the API (`"storage.create_annotation"` and `"storage.update_annotation"` jobs). This is because I want to create an alarm for if the jobs added by the API are piling up, but I don't want jobs added by admin pages to trigger that alarm. And and we might want to add the general sync-annotations queue length (including jobs added by admin pages) as another metric in future.

Reply:
https://github.com/hypothesis/h/blob/a58396c97e313141137a48bfcd1cc09c005c6c78/tests/h/services/search_index/_queue_test.py#L355-L372